modules/skybrowser/src/screenspaceskytarget.cpp

```diff
@@ -0,0 +1,430 @@
+#include <openspace/rendering/renderengine.h>
+#include <openspace/rendering/helper.h>
+#include <openspace/scene/scene.h>
+#include <openspace/camera/camera.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/opengl/programobject.h>
+
+namespace {
+    constexpr const char* _loggerCat = "ScreenSpaceSkyTarget";
+
+    constexpr const std::array<const char*, 7> UniformNames = {
+        "modelTransform", "viewProj", "showCrosshair", "showRectangle", "lineWidth", 
+        "dimensions", "lineColor"
+    };
+
+    constexpr const openspace::properties::Property::PropertyInfo CrosshairThresholdInfo =
+    {
+        "CrosshairThreshold",
+        "Crosshair Threshold",
+        "When the field of view is smaller than the crosshair threshold, a crosshair will"
+        "be rendered in the target."
+    }; 
+    
+    constexpr const openspace::properties::Property::PropertyInfo RectangleThresholdInfo =
+    {
+        "RectangleThreshold",
+        "Rectangle Threshold",
+        "When the field of view is larger than the rectangle threshold, a rectangle will"
+        "be rendered in the target."
+    };
+
+    constexpr const openspace::properties::Property::PropertyInfo AnimationSpeedInfo =
+    {
+        "AnimationSpeed",
+        "Animation Speed",
+        "The factor which is multiplied with the animation speed of the target."
+    };
+
+    constexpr const openspace::properties::Property::PropertyInfo AnimationThresholdInfo =
+    {
+        "AnimationThreshold",
+        "Animation Threshold",
+        "The threshold for when the target is determined to have appeared at its "
+        "destination. Angle in radians between the destination and the target position in"
+        "equatorial Cartesian coordinate system."
+    };
+
+    constexpr const openspace::properties::Property::PropertyInfo LineWidthInfo =
+    {
+        "LineWidth",
+        "Line Width",
+        "The thickness of the line of the target. The larger number, the thicker line."
+    };
+
+    struct [[codegen::Dictionary(ScreenSpaceSkyTarget)]] Parameters {
+        // [[codegen::verbatim(CrosshairThresholdInfo.description)]]
+        std::optional<float> crosshairThreshold;
+
+        // [[codegen::verbatim(RectangleThresholdInfo.description)]]
+        std::optional<float> rectangleThreshold;
+
+        // [[codegen::verbatim(AnimationSpeedInfo.description)]]
+        std::optional<double> animationSpeed;
+
+        // [[codegen::verbatim(AnimationThresholdInfo.description)]]
+        std::optional<float> animationThreshold;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<float> lineWidth;
+    };
+
+#include "screenspaceskytarget_codegen.cpp"
+
+} //namespace
+
+namespace openspace {
+    ScreenSpaceSkyTarget::ScreenSpaceSkyTarget(const ghoul::Dictionary& dictionary)
+        : ScreenSpaceRenderable(dictionary)
+        , _showCrosshairThreshold(CrosshairThresholdInfo, 4.f, 0.1f, 70.f)
```

Comment:
Are both the `_showCrosshairThreshold` and `_showRectangleThreshold` really needed? can't we just show one or the other? Seem to me like they could be combined into one threshold for when we switch from a crosshair to a rectangle? :)

Reply:
There is an inbetween state that show both the crosshair and the rectangle :)